diff -Naur a/data/io.github.qwersyk.Newelle.appdata.xml.in b/data/io.github.qwersyk.Newelle.appdata.xml.in
--- a/data/io.github.qwersyk.Newelle.appdata.xml.in	2023-06-30 14:01:39.464431135 +0530
+++ b/data/io.github.qwersyk.Newelle.appdata.xml.in	2023-06-30 14:03:34.438131935 +0530
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop">
 	<id>io.github.qwersyk.Newelle.desktop</id>
+	<icon type="stock">snap.newelle.io.github.qwersyk.Newelle</icon>
 	<metadata_license>CC0-1.0</metadata_license>
 	<project_license>GPL-3.0-or-later</project_license>
   <name>Newelle</name>
diff -Naur a/data/io.github.qwersyk.Newelle.desktop.in b/data/io.github.qwersyk.Newelle.desktop.in
--- a/data/io.github.qwersyk.Newelle.desktop.in	2023-06-30 14:02:00.711910427 +0530
+++ b/data/io.github.qwersyk.Newelle.desktop.in	2023-06-30 14:03:48.465927194 +0530
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=newelle
 Exec=newelle
-Icon=io.github.qwersyk.Newelle
+Icon=snap.newelle.io.github.qwersyk.Newelle
 Terminal=false
 Type=Application
 Categories=GTK;
diff -Naur a/src/gtkobj.py b/src/gtkobj.py
--- a/src/gtkobj.py	2023-07-03 15:22:45.610854183 +0530
+++ b/src/gtkobj.py	2023-07-03 15:26:51.803675518 +0530
@@ -151,7 +151,7 @@
     def run_python(self, widget):
         self.text_expander.set_visible(True)
         t = self.txt.replace("'", '"""')
-        process = subprocess.Popen(f"""flatpak-spawn --host python3 -c '{t}'""", stdout=subprocess.PIPE,
+        process = subprocess.Popen(f"""python3 -c '{t}'""", stdout=subprocess.PIPE,
                                    stderr=subprocess.PIPE, shell=True)
         stdout, stderr = process.communicate()
         text = "Done"
diff -Naur a/src/window.py b/src/window.py
--- a/src/window.py	2023-07-03 13:52:41.338569908 +0530
+++ b/src/window.py	2023-07-03 15:26:18.403094351 +0530
@@ -17,7 +17,7 @@
         self.main_program_block = Adw.Leaflet(fold_threshold_policy=True, can_navigate_back=True,
                                               can_navigate_forward=True,
                                               transition_type=Adw.LeafletTransitionType.UNDER)
-        self.path = ".var/app/io.github.qwersyk.Newelle/data"
+        self.path = "snap/newelle/common/data"
         if not os.path.exists(self.path):
             os.makedirs(self.path)
         self.filename = "chats.pkl"
@@ -753,8 +753,6 @@
     def execute_terminal_command(self, command):
         os.chdir(os.path.expanduser(self.main_path))
         console_permissions = ""
-        if not self.virtualization:
-            console_permissions = "flatpak-spawn --host "
         process = subprocess.Popen(console_permissions + '\n'.join(command), stdout=subprocess.PIPE,
                                    stderr=subprocess.PIPE, shell=True)
         stdout, stderr = process.communicate()
