diff -Naur a/data/io.github.qwersyk.Newelle.appdata.xml.in b/data/io.github.qwersyk.Newelle.appdata.xml.in
--- a/data/io.github.qwersyk.Newelle.appdata.xml.in	2023-06-30 14:01:39.464431135 +0530
+++ b/data/io.github.qwersyk.Newelle.appdata.xml.in	2023-06-30 14:03:34.438131935 +0530
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop">
 	<id>io.github.qwersyk.Newelle.desktop</id>
+	<icon type="stock">snap.newelle.io.github.qwersyk.Newelle</icon>
 	<metadata_license>CC0-1.0</metadata_license>
 	<project_license>GPL-3.0-or-later</project_license>
   <name>Newelle</name>
diff -Naur a/data/io.github.qwersyk.Newelle.desktop.in b/data/io.github.qwersyk.Newelle.desktop.in
--- a/data/io.github.qwersyk.Newelle.desktop.in	2023-06-30 14:02:00.711910427 +0530
+++ b/data/io.github.qwersyk.Newelle.desktop.in	2023-06-30 14:03:48.465927194 +0530
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=newelle
 Exec=newelle
-Icon=io.github.qwersyk.Newelle
+Icon=snap.newelle.io.github.qwersyk.Newelle
 Terminal=false
 Type=Application
 Categories=GTK;
diff -Naur a/src/window.py b/src/window.py
--- a/src/window.py	2023-07-03 13:52:41.338569908 +0530
+++ b/src/window.py	2023-07-03 13:54:59.170850985 +0530
@@ -17,7 +17,7 @@
         self.main_program_block = Adw.Leaflet(fold_threshold_policy=True, can_navigate_back=True,
                                               can_navigate_forward=True,
                                               transition_type=Adw.LeafletTransitionType.UNDER)
-        self.path = ".var/app/io.github.qwersyk.Newelle/data"
+        self.path = "snap/newelle/common/data"
         if not os.path.exists(self.path):
             os.makedirs(self.path)
         self.filename = "chats.pkl"
