diff -Naur a/data/org.gnome.newelle.appdata.xml.in b/data/org.gnome.newelle.appdata.xml.in
--- a/data/org.gnome.newelle.appdata.xml.in	2023-06-24 01:14:15.977033625 +0530
+++ b/data/org.gnome.newelle.appdata.xml.in	2023-06-24 01:15:49.405964652 +0530
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop">
 	<id>org.gnome.newelle.desktop</id>
+	<icon type="stock">snap.newelle.org.gnome.newelle</icon>
 	<metadata_license>CC0-1.0</metadata_license>
 	<project_license>GPL-3.0-or-later</project_license>
 	<description>
diff -Naur a/data/org.gnome.newelle.desktop.in b/data/org.gnome.newelle.desktop.in
--- a/data/org.gnome.newelle.desktop.in	2023-06-24 01:14:57.236449293 +0530
+++ b/data/org.gnome.newelle.desktop.in	2023-06-24 01:15:59.187522142 +0530
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=newelle
 Exec=newelle
-Icon=org.gnome.newelle
+Icon=snap.newelle.org.gnome.newelle
 Terminal=false
 Type=Application
 Categories=GTK;
